system/jlib/jdebug.cpp

```diff
@@ -935,24 +935,48 @@ static unsigned evalAffinityCpus()
 
 memsize_t getMapInfo(const char *type)
 {
+    // NOTE: 'total' heap value includes Roxiemem allocation, if present
     memsize_t ret = 0;
     VStringBuffer procMaps("/proc/%d/maps", GetCurrentProcessId());
-    VStringBuffer typeStr("[%s]", type);
     FILE *diskfp = fopen(procMaps.str(), "r");
     if (!diskfp)
         return false;
     char ln[256];
     while (fgets(ln, sizeof(ln), diskfp))
     {
-        if (strstr(ln, typeStr.str()))
+        bool skipline = true;
+        if ( !strcmp(type, "heap") ) // 'general' heap includes anon mmapped + sbrk
         {
-            unsigned __int64 addrLow, addrHigh;
-            if (2 == sscanf(ln, "%16" I64F "x-%16" I64F "x", &addrLow, &addrHigh))
+            if ( strstr(ln, "[heap") || (!strstr(ln, " /") && !strstr(ln, " [")) )
             {
-                ret = (memsize_t)(addrHigh-addrLow);
-                break;
+                if ( strstr(ln, " rw-p") || strstr(ln, " ---p") )
```

Comment:
Could do with some comments and an example /proc/../maps line, explaining exactly what this code is looking for here and why not " /" and not " [" and why these permissions etc.